base/reducedim.jl

```diff
@@ -151,12 +151,13 @@ has_fast_linear_indexing(a::Array) = true
 function check_reducedims(R, A)
     # Check whether R has compatible dimensions w.r.t. A for reduction
     #
-    # It returns an integer value value (useful for choosing implementation)
+    # It returns an integer value (useful for choosing implementation)
     # - If it reduces only along leading dimensions, e.g. sum(A, 1) or sum(A, (1, 2)),
     #   it returns the length of the leading slice. For the two examples above,
     #   it will be size(A, 1) or size(A, 1) * size(A, 2).
     # - Otherwise, e.g. sum(A, 2) or sum(A, (1, 3)), it returns 0.
     #
+    ndims(R) <= ndims(A) || throw(DimensionMismatch("Cannot reduce $(ndims(A))-dimensional array to $(ndims(R)) dimensions"))
```

Comment:
This is necessary to prevent a stackoverflow from `size_skip1` if `ndims(R) > ndims(A)`. I doubt that it used to be an error. I don't think there's any good reason to support it, but if others disagree, then `size_skip1` needs to be a stagedfunction.